### PR TITLE
Sean---LA132.industry_GCAM_USA

### PR DESCRIPTION
**Pull request by swd-turner**
#711

### CODE REVIEW

File: R/zchunk_LA132.Industry-gcamusa.R [hunk 1]
```diff
@@ -40,73 +43,165 @@ module_gcam.usa_LA132.Industry_DISABLED <- function(command, ...) {
...
+      # ^^ remove fuels that are inputs to cogen systems, i.e., not electricity
+      mutate(sector = "chp_elec") %>% rename(multiplier = value) %>%
+      # ^^ prepare for smooth join
+      left_join_error_no_match(filter(L123.in_EJ_R_indchp_F_Yh, GCAM_region_ID == gcam.USA_CODE),
+                               by = c("fuel", "year", "sector")) %>%
+      mutate(value = value * multiplier) %>%
+      select(-multiplier, -GCAM_region_ID) ->
+      L132.in_EJ_state_indchp_F
+
+    #Apportion nation-level industrial cogen output to states
```
Comment (bpbond) [inline R/zchunk_LA132.Industry-gcamusa.R]: Just because I need to say _something_: `# Apportion`

File: R/zchunk_LB126.Gas_ElecTD.R [hunk 2]
```diff
@@ -308,9 +308,9 @@ module_gcam.usa_LB126.Gas_ElecTD <- function(command, ...) {
       add_legacy_name("L126.out_EJ_state_td_elec") %>%
       add_precursors("L122.in_EJ_state_refining_F",
                      "L123.out_EJ_state_elec_F",
-                     "temp-data-inject/L132.in_EJ_state_indchp_F",
-                     "temp-data-inject/L132.in_EJ_state_indfeed_F",
-                     "temp-data-inject/L132.in_EJ_state_indnochp_F",
```
Comment (bpbond) [inline R/zchunk_LB126.Gas_ElecTD.R]: Thanks. Did you delete these files out of `inst/extdata/temp-data-inject/`?

**Comment by bpbond** [comment]
I'm getting all kinds of oldnew errors on this PR.

**Comment by swd-turner** [comment]
hmm weird... it tested clean on my system... will investigate tomorrow.

**Comment by swd-turner** [comment]
Don't think this is a dplyr version issue (I don't use `select("x","y")`). I just ran tests again on my machine and I am getting the oldnew fails. Guess I must have meddled with something after testing. On it now.

File: R/zchunk_LB126.Gas_ElecTD.R (hunk 2 above)
Comment (swd-turner) [reply R/zchunk_LB126.Gas_ElecTD.R]: Done now. I'd also left parts of the temp-inject code in the chunk, which was causing most of the oldnew fails.

**Comment by swd-turner** [comment]
@bpbond FYI had to solve a weird test fail here for output 'L132.in_EJ_state_indeed_F.csv'. Turns out that if the first 1000 entries to the value column zero, the `read_csv` call in the oldnew test (#52) will fail. The reason is that the function looks at the first 1000 entries to determine whether it's integer or double... so there will be parsing fails if it assumes integer and then doubles appear (https://github.com/tidyverse/readr/issues/645). The problem was solved by re-arranging the tibble to bring some non-zero values to the top.

**Comment by swd-turner** [comment]
ok I now see that this is a recognized issue... should have checked the common errors first. Doh!

**Comment by bpbond** [comment]
>ok I now see that this is a recognized issue... should have checked the common errors first. Doh!

haha! Well, kudos for figuring it out.

FYI another solution is to use FLAG_PROTECT_FLOAT. Yeah, it's a crappy fail and not our fault.

**Comment by bpbond** [comment]
Waiting on @gokuliyer review.

**Pull request merged**